feat/Walls/ReportRemon

SamehAmnoun opened the pull request.

SamehAmnoun committed:
fix: ReportSameh

SamehAmnoun committed:
fix: player passing through the wall diagonally

-Changed else if to if in dontMove()
-Moved shouldNtMove functions to the Wall class
-Changed the shouldNtMove functions to avoid duplicated code

SamehAmnoun committed:
-workspace.xml

SamehAmnoun committed:
Merge branch 'master'

fix: the Said Bug

-Changed the dontMove() approach in gameViewManager to move() in the Player Class.

SamehAmnoun committed:
refactor: renamed ShouldNtMove functions to canMove

feat: canMove functions now use dynamic wall's size instead of hardcoded numbers

SamehAmnoun committed:
refactor: walls constant

SamehAmnoun committed:
refactor: merged unnecessary nested ifs

MohamedSaidSallam commented on src/view/GameViewManager.java:
```diff
@@ -106,8 +106,10 @@ public void handle(long now) {
     private void gameUpdate() {
         LevelManager.createEnemies();
         LevelManager.createObstacles();
+        LevelManager.createWall();
```
This line was the problem
Credit to @rameziophobia

MohamedSaidSallam replied:
it should be in game start

SamehAmnoun committed:
fix: New wall created every 0.0016sec which caused lag

-createWall is now called in gameStart instead of gameUpdate
-checkCollision_wall is now active
-changed the implementation of isIntersects() to getBoundsInParent().intersects(s.getBoundsInParent());
